src/System.Diagnostics.Metrics.Temp/CounterFunc.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+#nullable enable
+
+namespace System.Diagnostics.Metrics
+{
+    public class CounterFunc<T> : ObservableMeterInstrument<T> where T : unmanaged
```

Comment:
I think this is called `ObservableCounter` (or do you intend to change it later)?

Reply:
yes will be change to match https://github.com/dotnet/designs/pull/211/files. This PR is to get started with a temp proj file.